File: Step 03 Arrays/Step 3.2 Medium/Pair With Max Sum/Pair With Max Sum.cpp
class Solution {
  public:
    long long pairWithMaxSum(long long arr[], long long N) {
        // Your code goes here
        long long sum = LONG_MIN;
        long long maxi = LONG_MIN;
        for (int i=1; i<N; ++i) {
            sum = arr[i] + arr[i-1];
            maxi = max(maxi, sum);
        }
        return maxi;
    }
};

File: Step 03 Arrays/Step 3.2 Medium/Pair With Max Sum/Pair With Max Sum.java

//User function Template for Java

class Solution {
    
    public static long pairWithMaxSum(long arr[], long N)
    {
        // Your code goes here
        long sum = Long.MIN_VALUE;
        long max = Long.MIN_VALUE;
        for (int i = 0; i < N-1; i++) {
            sum = arr[i] + arr[i+1];
            if(sum > max) {
                max = sum;
            }
        }
        return max;
    }
}